src/components/chip/index.tsx

```diff
@@ -292,6 +292,13 @@ const Chip = ({
           return {
             marginRight: Spacings.s2
           };
+        case 'iconSource':
+          if (label || leftElement || rightElement) {
+            return {
+              marginLeft: Spacings.s1,
+              marginRight: Spacings.s2
+            };
```

Comment:
overall It looks much better. 
But I think it mostly solve the right icon
notice that the use can either pass `rightIconSource` or `iconSource` (rendered on the left)
I think those margins works good for right icons and maybe should be opposite for left icon